src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/logging/LazyToStr.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package za.co.absa.hyperdrive.trigger.scheduler.utilities.logging
+
+/**
+ *  Class deferring computation on toString call.
+ *
+ *  @param value That should be computed only after toString call.
+ *  @tparam A Type of value computation result
+ */
+class LazyToStr[A](value: => A) {
```

Comment:
This might not be required anymore. The `scala-logging` macro should wrap the call to in
```scala
if (logger.is<Level>Enabled) logger.<level>(message, args)
```

So proper usage should be
```scala
logger.debug("Data = ${heavyComputationOfData()}")
```
However, we need first to check that it behaves this way.

Reply:
So I checked it out, and it can be replaced directly with `logger.debug(s"... ${value}")`